File: Algorithms/0128_Longest_Consecutive_Sequence/Java/LongestConsecutiveSequenceSolution1.java

// Space: O(n)
// Time: O(n)

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length <= 1) return nums.length;
        List<Integer> cache = new ArrayList(Arrays.asList(nums));

        int[] cache1 = cache.stream().distinct().sorted().mapToInt(Integer::intValue).toArray();

        int res = 0, counter = 1, index = 1;

        while (index < nums.length) {
            if (nums[index] == nums[index - 1] + 1) {
                counter += 1;
            }
            else {
                res = Math.max(res,counter);
                counter = 1;
            }
            index+=1;
        }
        return Math.max(res,counter);
    }
}

File: Algorithms/0128_Longest_Consecutive_Sequence/Python/Longest_Consecutive_Sequence_Solution_1.py

# Space: O(n)
# Time: O(n)

class Solution:
    def longestConsecutive(self, nums):
        if not nums: return 0
        nums = sorted(list(set(nums)))
        res = 0
        counter = 1
        index = 1
        while index < len(nums):
            if nums[index] == nums[index - 1] + 1:
                counter += 1
            else:
                res = max(res, counter)
                counter = 1
            index += 1

        return max(res, counter)




